src/pages/settings/Wallet/revealCardDetailsUtils.ts

```diff
@@ -0,0 +1,57 @@
+type State = {
+    details: {
+        pan: string;
+        expiration: string;
+        cvv: string;
+        privatePersonalDetails: {
+            address: {
+                street: string;
+                street2: string;
+                city: string;
+                state: string;
+                zip: string;
+                country: string;
+            };
+        };
+    };
+    loading: boolean;
+    error: string;
+};
+
+type Action = {type: 'START'} | {type: 'SUCCESS'; payload: State['details']} | {type: 'FAIL'; payload: string};
```

Comment:
nit, Id move 'START', 'SUCCESS', 'FAIL' to some kind of const variable

Reply:
it's TS typed, so the right values should be autocompleted by the IDE, hence no constants needed imo.

Reply:
Card page is in .js tho. Im not sure if any checks happen there when dispatching actions